designs/air-gapped.md

```diff
@@ -0,0 +1,97 @@
+* Populate disconnected environments with dependencies with Internet connection.
+ 
+## Current state
+In order to support disconnected environments, users currently have to:
+1. Provision an OCI registry.
+1. Run `eksctl anywhere download artifacts` to download manifests to disk.
+1. Run `eksctl anywhere download images` to download images and charts to disk.
+1. Run `eksctl anywhere import images` to push container images and charts to a registry.
+1. Configure their cluster spec file to point to the registry 
+1. Run all cluster commands (`create`, `upgrade`, etc.) with the `--bundles-override` flag pointing to the `Bundles` manifest downloaded in the first step.
+
+## Overview of Solution
+1. Users will provide an OCI registry.
+1. We will store all dependencies (container images, helm charts and yaml manifests) in that registry.
+1. If a cluster config with a [`registryMirrorConfiguration`](https://github.com/aws/eks-anywhere/blob/main/pkg/api/v1alpha1/cluster_types.go#L51) is provided, the CLI and the rest of components will always pull them from it.
+No extra flags/configuration will be needed and the user will interact with this cluster as with any other one.
+
+## Solution Details
+
+### Populating the registry
+
+In order to populate the registry, we will offer two commands:
+```sh
+eksctl anywhere export artifacts --output artifacts.tar
+eksctl anywhere import artifacts --input artifacts.tar --registry myregistry.com
+```
+
+The first one downloads all dependencies (yaml manifests, images and helm charts). The only command argument here is the destination file. This command will create a tarball containing all 3 types of artifacts in that location.
+The second one unpackages the tarball created in the first command, reads the packaged `Bundles` manifest and imports the referenced dependencies to a registry. The input here will be the artifacts tarball, the registry endpoint and the registry credentials (provided through env vars).
+
+This two commands can be expanded in the future to add more capabilities like selectively exporting dependencies (eg. only for one kubernetes version or for only one provider).
+
+### Pre packaged dependencies
+
+An specific version of the CLI tied to one `Bundles` manifest will always produce the exact same artifacts tarball.
+This means that we can prepackage the dependencies, store them in a public bucket and reference the tarball in the `Release` manifest.
+This simplifies the experience for users who are interested in the default dependencies bundle.
+
+## Implementation
+
+The proposed design can be implemented in 4 incremental phases:
+1. Add the two new commands but only push images and charts to the registry. This will require to keep using the `--bundles-override` pointing to a `Bundles` manifest in disk. This is an incremental improvement over the current state of `download images` so work should be minimal.
+2. Push manifests to OCI registry and add the capability to the CLI to download them.
```

Comment:
@abhay-krishna this is something interesting we should test out to see our options. https://oras.land/cli/1_pushing/ could be interesting for this.